### ISSUE TITLE
Working with Image fields

### ISSUE THREAD

**Issue by PathToSharePoint** [pnp/pnpjs#1612]
### Category
- [ ] Enhancement
- [ ] Bug
- [x] Question
- [ ] Documentation gap/issue

### Version

Please specify what version of the library you are using: [latest]

Please specify what version(s) of SharePoint you are targeting: [online]

### Question

I am considering using a Microsoft Lists field of type "Image" to store images, instead of using a document library or list attachments. Is working with Image fields covered by PnPjs, and if so would you have an example to share?

**Comment by jimmywim** [comment]
AFAIK, when you upload an image to an "Image" field type, it goes into SiteAssets of the site hosting the List. The field itself stores a JSON object that includes the full URL to the uploaded file.

**Comment by ValerasNarbutas** [comment]
@PathToSharePoint 
new image field in sharepoint online holds json like:
```json
{
    "type":"thumbnail",
    "fileName":"uploadedImageFileName.jpg",
    "nativeFile":{},
    "fieldName":"ImageFld",
    "serverUrl":"https://tenant.sharepoint.com",
    "serverRelativeUrl":"/sites/siteName/SiteAssets/Lists/76605040-a746-411b-b50b-a0a0d39a8bd8/uploadedImageFileName.jpg",
    "id":"9df684cf-f882-4d66-aed3-7ef0508e454c"
    }
```

Image field upload action creates folder "Lists" and subfolder with name of your list Id, when uploads file to it and adds path in json  as mentioned by @jimmywim 
PnpJs currently does not specifically "work" with image fields, but using create field by schema might do a trick:
https://pnp.github.io/pnpjs/sp/fields/#create-a-field-using-an-xml-schema

just need to grab xml schema for image field and try to apply using this function.
Hope this helps.

@patrick-rodgers you added help wanted tag, you think this would be good to have as function for image fields?

**Comment by juliemturner** [comment]
To your point if the library doesn't have a shortcut, you can use the field using the XML schema, and I often need to do this for special cases... the `help wanted` tag is meant that in general the maintainers focus our attention to questions/issues with the library itself, if something is tagged as `coding request`/`help wanted` it means we feel like someone is looking for help writing code for something that is doable and not a bug/enhancment to PnPjs... 

That said yes if Image field is a field type that is supported by any type of "list", we could certainly look to enhance the library to include a specialized function to create an image field. What I wouldn't want us to do is create a specialized function if a field type is so special that it's only available in certain scenarios, which doesn't seem to be the case with Image but I can't say I've used the field type extensively.

**Comment by jimmywim** [comment]
Further to this, I reckon the library should definitely provide a strong way of _reading_ Image field types, as well as assist in the creation of them.

**Comment by juliemturner** [comment]
Please feel free to create enhancement requests that outline what you think should be added.

**Comment by ValerasNarbutas** [comment]
@juliemturner i can try to add this to pnpjs

**Comment by patrick-rodgers** [comment]
Added addImageField to the fields module and setImageField to item allowing you to easily add image fields to a list and then set the value of that field. Will be released in 2.8

**Comment by ValerasNarbutas** [comment]
Superb! thanks @patrick-rodgers
